lib/src/sql/statements/foreach.rs

```diff
@@ -0,0 +1,82 @@
+use crate::ctx::Context;
+use crate::dbs::{Options, Transaction};
+use crate::doc::CursorDoc;
+use crate::err::Error;
+use crate::sql::block::{block, Block};
+use crate::sql::comment::{mightbespace, shouldbespace};
+use crate::sql::error::IResult;
+use crate::sql::param::{param, Param};
+use crate::sql::value::{value, Value};
+use derive::Store;
+use nom::bytes::complete::tag_no_case;
+use revision::revisioned;
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Display};
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize, Store, Hash)]
+#[revisioned(revision = 1)]
+pub struct ForeachStatement {
+	pub param: Param,
+	pub range: Value,
+	pub block: Block,
+}
+
+impl ForeachStatement {
+	/// Check if we require a writeable transaction
+	pub(crate) fn writeable(&self) -> bool {
+		self.range.writeable() || self.block.writeable()
+	}
+	/// Process this type returning a computed simple Value
+	pub(crate) async fn compute(
+		&self,
+		_ctx: &Context<'_>,
+		_opt: &Options,
+		_txn: &Transaction,
+		_doc: Option<&CursorDoc<'_>>,
+	) -> Result<Value, Error> {
+		Err(Error::FeatureNotYetImplemented {
+			feature: "FOR statements",
+		})
```

Comment:
Aren't new syntax additions beyond the scope of initial versioning? Feels like this should be a separate PR.